### ISSUE TITLE
Roku (and only Roku) delivering empty playlist.

### ISSUE THREAD

**Issue by TrevorWilsonVO** [dtankdempse/free-iptv-channels#26]
For some reason, I seem to be getting an empty playlist for Roku, and only Roku. Every other source provides the channels for that service, but the Roku one contains nothing but the link to the guide data. The JSON file on i.mjh.nz seems to be working fine, but whether it be the hosted script link on the GitHub page or my own Docker image, the Roku playlist is always empty. This seems to be a relatively recent issue, as the Roku playlist worked fine for me this morning. Thank you for all the work you put into this project, let me know if there's any further information I should provide, and I hope this issue is relatively easy to fix!
![Screenshot 2024-12-05 at 4 59 32 PM](https://github.com/user-attachments/assets/da28b4e3-8e0d-4cd7-8b82-c80e6806b46c)

**Comment by Fofer** [comment]
I came here to report the exact same issue. Thanks!

**Comment by widdlepear** [comment]
Roku and Stirr both are returning code 404.

**Comment by dtankdempse** [comment]
The issue appears to be on Matt's server. It isn't returning any data when the JSON file is requested:

https://i.mjh.nz/Roku/.channels.json

The compressed file is empty as well:
https://github.com/matthuisman/i.mjh.nz/blob/master/Roku/.channels.json.gz

The returned JSON contains only an empty structure:
`{"channels":{},"headers":{"user-agent":"rokuandroid"}}`

**Comment by Fofer** [comment]
> The issue appears to be on Matt's server. It isn't returning any data when the JSON file is requested:

Thanks for the explanation, that helps... I see it's also being discussed [here](https://github.com/matthuisman/i.mjh.nz/issues/127#issuecomment-2524979763).  Hopefully gets resolved soon. Happy holidays ahead.

**Comment by matthuisman** [comment]
Roku removed the domain name I was using for the playlists.
Ill need to MITM the latest android app and figure out new urls to use.
Not sure on ETA sorry

**Comment by matthuisman** [comment]
most channels (all?) should be back now

**Comment by Jag4343** [comment]
> most channels (all?) should be back now

**Comment by Jag4343** [comment]
I’m using the sane url I have been using for a few months and I am still not getting any channels. I’m using it on Tivimate and getting a HttpDataSource Exception error. Is there anyway to fix it? Thanks

**Comment by Fofer** [comment]
Same here, Roku's channels and EPG appear but none of them actually play.  They were working after the fix last week, but stopped working again shortly thereafter.

**Comment by clseibold** [comment]
Afaik, the roku playlist is no longer empty, but the streams don't work because the script is getting an undefined for `mjh_url` since the i.mjh.nz roku channels.json file doesn't provide the `mjh_url` property.

**Comment by matthuisman** [comment]
whoever works on this project needs to the same thing for roku method that they did for Pluto etc

**Comment by dtankdempse** [comment]
> whoever works on this project needs to the same thing for roku method that they did for Pluto etc https://github.com/dtankdempse/free-iptv-channels/blob/main/node/index.js#L131 but rok-
> 
> @dtankdempse @clseibold

Great! Thanks for the reply.  I'll have this changed and update the files.

**Comment by clseibold** [comment]
Hopefully we can find a better source than matthuisman's stuff - they are an extremely disrespectful developer, and they don't seem that dedicated to fixing issues that people have.

**Comment by dtankdempse** [comment]
> Hopefully we can find a better source than matthuisman's stuff - they are an extremely disrespectful developer, and they don't seem that dedicated to fixing issues that people have.

I'm not sure what you mean by that. Matt has shared and done a lot for the community.

**Comment by matthuisman** [comment]
keep pushing and the json files and the redirect urls will disappear and everyone can thank you :)

**Comment by clseibold** [comment]
matthuisman is not a God. This **diva** attitude that he needs to get the upmost respect and for everybody to stop wasting his time, as if his time is more precious than everybody else's is childish.

**Comment by dtankdempse** [comment]
@matthuisman Thanks for everything you do, bro. Don’t let one person’s opinion push you into making any drastic changes. The files and code you share with us are appreciated.

**Comment by matthuisman** [comment]
https://github.com/matthuisman/i.mjh.nz/issues/152 is where I was so rude and "Godlike"

@clseibold 
im sure everyone will be keen to see your scraper.
There must be a stream url as it plays in browser right?
Go get it! Chrome webtools will show it.
So easy!!

**Comment by clseibold** [comment]
No, matt. Where you were "godlike" is when you started threatening to remove all your files, as if it's some major devastation, and you continue to act this way by implying you're the only person who can program this stuff to a person you barely know. Like I said... big ego.

**Comment by matthuisman** [comment]
nope. you just came in with little knowledge and think its easy.
So you go prove that please :)
Then ill eat my hat

**Comment by matthuisman** [comment]
Looks fixed now:
https://tinyurl.com/multiservice21?service=Roku

**Comment by clseibold** [comment]
Nah, that's not what happened. You gave **incorrect** details as to why a channel was not listed, suggesting that it's "not currently possible" to scrape it. You did not mention anything about widevine or other such. This tells me just like how some of your stuff is closed, you want to keep details hush-hush, which is pretty consistent with acting like your the only person with the knowledge of doing this stuff.
At the very outset you were rude, because clearly you do not know how to speak to users. When someone asks you why there's no channel, you specify exactly why there's no channel. You don't tell them a snarky response like "You'll need to figure out the url yourself :)".

**Comment by dtankdempse** [comment]
> Looks fixed now: https://tinyurl.com/multiservice21?service=Roku

 I fixed it after you replied. I’ll take care of the index.js file later when I have some time.

**Comment by clseibold** [comment]
@dtankdempse Thanks for updating this! I tried [telling you](https://github.com/dtankdempse/free-iptv-channels/issues/27#issuecomment-2567665900) this is what you needed to do, but you didn't listen for some reason, lol.

**Comment by dtankdempse** [comment]
> @dtankdempse Thanks for updating this! I tried [hinting to you](https://github.com/dtankdempse/free-iptv-channels/issues/27#issuecomment-2567665900) that this is what you needed to do, but you didn't listen for some reason, lol. Anyways, it's fixed now, so a lot of us users will be glad.

This is not what you tried telling me. Your words:

"imjh no longer provides m3u urls for anything, and it looks like this repository's code is still trying to use mjh's Roku channels list as if it still provided these m3u urls (mjh_url is undefined). @dtankdempse The handleRoku function needs to be updated to account for this. However, I don't know if there's any other service that can be used for linking the playlists."

Sounds as if you were thinking this script relied on Matt's old setup where he listed the M3U URLs. This code was changed since then and has worked after that change. So no, this is not what you described.

**Comment by matthuisman** [comment]
@dtankdempse i suggest closing this issue now that its fixed and also locking it :)
